### PR TITLE
First Fully Working Prototype

### PR DESCRIPTION
**Pull request by lighterm**

### CODE REVIEW

File: packages/handler/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,9 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "example-handler",
+    srcs = ["main.go"],
+    deps = ["@io_bazel_rules_go//go/tools/coverdata"],
+    importpath = "example-library",
+    visibility = ["//visibility:public"],
+)
```
Comment (pratyai) [inline packages/handler/BUILD.bazel]: Could you format all the build files with the buildifier, please? Some small stuff like ending file with a new line will be changed then I think.

File: packages/handler/main.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/bazelbuild/rules_go/go/tools/coverdata"
+)
+
+// Function collect writes the data in coverdata to "coverage.out" file
+var collect func() = func() {
+
```
Comment (pratyai) [inline packages/handler/main.go]: Please drop the leading and trailing newlines inside the function body.

File: packages/handler/main.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+func init() {
+	fmt.Println("-- Coverage Collection is started --")
+	LastCallForFunccoverReport = &collect
+	go periodicalCollect()
+}
+
+// Function periodicalCollect calls the collect function every 500ms
+func periodicalCollect() {
+
+	duration, _ := time.ParseDuration("500ms")
```
Comment (pratyai) [inline packages/handler/main.go]: duration = 500 * time.Millisecond

**Review by pratyai — CHANGES_REQUESTED** [review]

File: packages/handler/BUILD.bazel (hunk 1 above)
Comment (lighterm) [reply packages/handler/BUILD.bazel]: Thanks, fixed it.

**Comment by lighterm** [comment]
I changed the code_coverage_mode option. Now it is in the rules_go repo instead of Bazel. Other option is still in the Bazel repo so we still need to install a modified Bazel to use it.  Details are in the document. 

I also did some changes in the structure of the project.

There are new commits in the rules_go and bazel repos as well.

**Pull request merged**